Improve error message when git is not found

ipenas-cl opened the pull request:
Previously when git was not available in PATH, users would see a generic 'Program git not found in the tree or in PATH' error. This change adds a helpful hint explaining why git is needed and how to resolve the issue.

The improved error now explains that git is required for:
- Version information in 'dune subst'
- Build info and artifact substitution
- Package management operations

Addresses #10393

Alizter commented on src/dune_vcs/vcs.ml:
```diff
@@ -55,7 +55,16 @@ let git, hg =
+           match prog with
+           | "git" ->
+             Some
+               "Git is required for version information in 'dune subst', build info, \
+                and package management. Install git or add it to your PATH."
```
Current ocamlformat wraps this slightly differently:

```suggestion
               "Git is required for version information in 'dune subst', build info, and \
                package management. Install git or add it to your PATH."

```

Alizter commented on src/source/workspace.ml:
```diff
@@ -575,11 +577,14 @@ module Context = struct
-        let name = Option.value ~default name in
+        let name = 
+          match name with
+          | Some (loc, name) -> name
```
This is the reason for the build errors in CI, due to the unused `loc`.

I am not quite sure what you are intending to do with this extra location?

ipenas-cl replied:
You're right! I see the issue now. The `located` captures both location and value, but I'm only using the name and discarding the location.

  I think the better approach would be to use the location when parsing fails. Let me update this to actually use the location for better error messages. Would something like this make more
  sense?

  ```ocaml
  let name =
    match name with
    | Some (loc, name) ->
      (try Context_name.parse_string_exn (loc, name)
       with exn -> User_error.raise ~loc [...])
    | None -> default

  Or should I remove the located wrapper entirely if we're not using the location?

Leonidas-from-XIV commented on src/source/workspace.ml:
```diff
@@ -564,7 +564,9 @@ module Context = struct
-        field_o "name" (Dune_lang.Syntax.since syntax (1, 10) >>> Context_name.decode)
+        field_o
+          "name"
+          (located (Dune_lang.Syntax.since syntax (1, 10) >>> Context_name.decode))
```
I do not understand this change at all. If you introduce a location here and then drop it in the bottom you could just... not do that?

Alizter commented:
@ipenas-cl This is looking good now. Could you squash and rebase on top of `main` and force push here?

ipenas-cl committed:
Improve error message when git is not found

Previously when git was not available in PATH, users would see a generic
'Program git not found in the tree or in PATH' error. This change adds
a helpful hint explaining why git is needed and how to resolve the issue.

The improved error now explains that git is required for:
- Version information in 'dune subst'
- Build info and artifact substitution
- Package management operations

Addresses #10393

Signed-off-by: Ignacio Peña <ignacio.pena87@gmail.com>

Alizter commented:
@Leonidas-from-XIV I don't think its necessary to add a cram test here. Setting up `PATH` scrubbing the the cram test is already going to be quite involved. As it stands this PR improves the hint on an existing error, so I think its fine to just merge.
